instrumentation/jms-1.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jms/JmsMessageConsumerInstrumentation.java

```diff
@@ -57,18 +55,13 @@ public static long onEnter() {
 
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void stopSpan(
-        @Advice.This MessageConsumer consumer,
         @Advice.Enter long startTime,
         @Advice.Return Message message,
         @Advice.Thrown Throwable throwable) {
       MessageDestination destination;
       if (message == null) {
-        destination =
-            InstrumentationContext.get(MessageConsumer.class, MessageDestination.class)
-                .get(consumer);
-        if (destination == null) {
-          destination = MessageDestination.UNKNOWN;
-        }
+        // Do not create span when no message is received
+        return;
       } else {
```

Comment:
Not a huge deal, but my preference is to remove the redundant `else` when there is a return above it.

Reply:
fixed